### PR TITLE
Header

### LINKED ISSUE #864: Header component?
It would be nice if the header (https://github.com/primer/css/tree/master/src/header) could be added in as a component, along with the header items.

### PR DESCRIPTION
**Pull request by pavelkeyzik**
Implements the Header component from `primer/css`. Please, let me know if something is wrong with my implementation. I have two questions about my implementation.

### Questions 1

Are changes in the `docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js` file okay? I moved the line `...doctocatComponents` above because when I was using the `Header` component in the documentation it was rendering the gatsby-theme's component instead of the primer's component.

### Question 2

Do we need to implement the `selected` property for `Header.Link` component?

Closes #864

### Screenshots
image

### Merge checklist
- [x] Added or updated TypeScript definitions (`index.d.ts`) if necessary
- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge

### CODE REVIEW

**Comment by emplums** [comment]
Thanks for the contribution & sorry for the late reply! I'll take a look at this early next week

File: src/Header.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+})`
+  font-weight: ${get('fontWeights.bold')};
+  color: ${get('colors.text.white')};
+  white-space: nowrap;
+  cursor: pointer;
+  text-decoration: none;
+  display: flex;
+  align-items: center;
+
+  /* TODO: Add props.selected styles */
```
Comment (emplums) [inline src/Header.js:60]: You can remove this todo comment for now

**Review by emplums — APPROVED** [review]
Wow nice work @pavelkeyzik! This is exactly how I would have implemented the component as well. I don't have much to add - we can hold off on adding selected styles since Primer CSS isn't currently implementing those either it looks like. Just left one comment about removing a comment and then this should be good to merge into the release branch! :) Thanks for your contribution!

File: src/Header.js (hunk 1 above)
Comment (pavelkeyzik) [reply src/Header.js]: Okay 😊

**Comment by pavelkeyzik** [comment]
@emplums Thanks for the kind words 🥰

**Pull request merged**